pkg/messaging/brokers/brokers_nats.go

```diff
@@ -0,0 +1,42 @@
+//go:build !rabbitmq
+// +build !rabbitmq
+
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package brokers
+
+import (
+	"log"
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/pkg/messaging/nats"
+)
+
+// SubjectAllChannels represents subject to subscribe for all the channels.
+const SubjectAllChannels = "channels.>"
+
+type (
+	Publisher nats.Publisher
+	PubSub    nats.PubSub
+)
```

Comment:
Let's simplify remove these wrappers and update the base interface with `Close`:
```go
// Publisher specifies message publishing API.
type Publisher interface {
	// Publishes message to the stream.
	Publish(topic string, msg Message) error

	// Close gracefully closes message publisher's connection.
	Close() error
}

// MessageHandler represents Message handler for Subscriber.
type MessageHandler interface {
	// Handle handles messages passed by underlying implementation.
	Handle(msg Message) error

	// Cancel is used for cleanup during unsubscribing and it's optional.
	Cancel() error
}

// Subscriber specifies message subscription API.
type Subscriber interface {
	// Subscribe subscribes to the message stream and consumes messages.
	Subscribe(id, topic string, handler MessageHandler) error

	// Unsubscribe unsubscribes from the message stream and
	// stops consuming messages.
	Unsubscribe(id, topic string) error

	// Close gracefully closes message subscriber's connection.
	Close() error
}
```
and use `messaging.{Publisher,Subscriber,PubSub}` everwhere instead of wrapped versions (`brokers.PubSub`, `nats.PubSub`, `mqtt.PubSub`...).